Framework/PythonInterface/plugins/algorithms/FitIncidentSpectrum.py

```diff
@@ -0,0 +1,221 @@
+                '\"[Start],[Increment],[End]\". If empty use default binning. The calculated '
+                'spectrum will use this binning')
+
+        self.declareProperty(
+            name='BinningForFit',
+            defaultValue='',
+            doc='Bin range for fitting given as a comma separated string in the format '
+                '\"[Start],[Increment],[End]\". If empty use BinningForCalc. The '
+                'incident spectrum will be rebined to this range before being fit.')
+
+        self.declareProperty(
+            name='FitSpectrumWith',
+            defaultValue='GaussConvCubicSpline',
+            validator=StringListValidator(['GaussConvCubicSpline', 'CubicSpline', 'CubicSplineViaMantid']),
+            doc='The method for fitting the incident spectrum.')
+
+    def validateInputs(self):
+        issues = dict()
+
+        input_ws = self.getProperty('InputWorkspace').value
+        range_min = input_ws.readX(0)[0]
+        bin_width = input_ws.readX(0)[1] - input_ws.readX(0)[0]
+        range_max = input_ws.readX(0)[-1]
+        binning_for_calc = self.getProperty('BinningForCalc').value
+        if not binning_for_calc == "":
+            parsed_calc = self.parse_binning(binning_for_calc)
+            if parsed_calc[0] < range_min - bin_width:
+                issues['BinningForCalc'] = "Lower bin range outside of InputWorkspace range, minimum value: %f" % (
+                    range_min - bin_width)
+            if parsed_calc[-1] > range_max + bin_width:
+                issues['BinningForCalc'] = "Upper bin range outside of InputWorkspace range, maximum values: %f" % (
+                    range_max + bin_width)
+            if parsed_calc.size == 0:
+                issues['BinningForCalc'] = "Invalid parameters for bin range"
+        binning_for_fit = self.getProperty('BinningForFit').value
+        if not binning_for_fit == "":
+            parsed_fit = self.parse_binning(binning_for_fit)
+            if parsed_fit[0] < range_min - bin_width:
+                issues['BinningForFit'] = "Lower bin range outside of InputWorkspace range, minimum value: %f" % (
+                    range_min - bin_width)
+            if parsed_fit[-1] > range_max + bin_width:
+                issues['BinningForFit'] = "Upper bin range outside of InputWorkspace range, maximum values: %f" % (
+                    range_max + bin_width)
+            if parsed_calc.size == 0:
+                issues['BinningForFit'] = "Invalid parameters for bin range"
+        return issues
+
+    def _setup(self):
+        self._input_ws = self.getProperty('InputWorkspace').value
+        self._output_ws = self.getProperty('OutputWorkspace').valueAsStr
+        self._binning_for_fit = self.getProperty('BinningForFit').value
+        self._fit_spectrum_with = self.getProperty('FitSpectrumWith').value
+        self._binning_for_calc = self.getProperty('BinningForCalc').value
+        if self._binning_for_calc is "":
```

Comment:
It is generally considered better in Python to test for emptyness by just: 
`if not self._binning_for_calc:`